### PR TITLE
IW-1226 | Custom smart banner

### PR DESCRIPTION
**Pull request by rybmat**
## Links

* http://wikia-inc.atlassian.net/browse/IW-1226
* https://github.com/Wikia/app/pull/16338

## Reviewers

@Wikia/iwing

### CODE REVIEW

File: app/components/application-wrapper.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ export default Component.extend({
   classNameBindings: [
     'smartBannerVisible',
     'isFandomAppSmartBannerVisible:with-fandom-app-smart-banner',
+    'isCustomSmartBannerVisible:with-fandom-app-smart-banner',
```
Comment (latata) [inline app/components/application-wrapper.js]: how about renaming this class to `with-smart-banner`?

File: app/components/custom-smart-banner.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+  classNames: ['fandom-app-smart-banner'],
+  dayInMiliseconds: 86400000,
+  // sync with scss variable $fandom-app-smart-banner-height
+  fandomAppSmartBannerHeight: 85,
+  trackCategory: 'custom-smart-banner',
+
+  closeButtonSelector: '.fandom-app-smart-banner__close',
+
+  text: computed('wikiVariables.smartBannerAdConfiguration.text', function () {
```
Comment (latata) [inline app/components/custom-smart-banner.js]: can't we use `oneWay` here?

Comment (latata) [reply app/components/custom-smart-banner.js]: I would also create another computed property which stores `wikiVariables.smartBannerAdConfiguration`. Not to repeat `wikiVariables.smartBannerAdConfiguration` for all the properties

File: app/services/smart-banner.js [hunk 3]
```diff
@@ -9,25 +11,65 @@ export default Service.extend({
...
-  trackCategory: 'fandom-app-smart-banner',
+  fandomAppCookieName: 'fandom-sb-closed',
+  customCookieName: 'custom-sb-closed',
 
   dbName: readOnly('wikiVariables.dbName'),
   isUserLangEn: equal('currentUser.language', 'en'),
   shouldShowFandomAppSmartBanner: and('isUserLangEn', 'wikiVariables.enableFandomAppSmartBanner'),
-  isFandomAppSmartBannerVisible: and('shouldShowFandomAppSmartBanner', 'smartBannerVisible'),
+  isFandomAppSmartBannerVisible: computed('shouldShowFandomAppSmartBanner', 'smartBannerVisible', function () {
+    return this.get('shouldShowFandomAppSmartBanner')
```
Comment (latata) [inline app/services/smart-banner.js]: we don't need to use getters now

File: app/services/smart-banner.js [hunk 4]
```diff
@@ -9,25 +11,65 @@ export default Service.extend({
...
+  }),
+  isCustomSmartBannerVisible: and(
+    'shouldShowFandomAppSmartBanner',
+    'smartBannerVisible',
+    'wikiVariables.smartBannerAdConfiguration.text',
+    'isInCustomSmartBannerCountry',
+    'isSystemTargetedByCustomSmartBanner',
+  ),
+
+  isInCustomSmartBannerCountry: computed('wikiVariables.smartBannerAdConfiguration.countries', function () {
```
Comment (latata) [inline app/services/smart-banner.js]: let's create computed property for `wikiVariables.smartBannerAdConfiguration`

File: app/services/smart-banner.js [hunk 5]
```diff
@@ -36,21 +78,21 @@ export default Service.extend({
...
 
   /**
-  * @param {string} action
-  * @returns {void}
-  */
-  track(action) {
+   * @param {string} action
+   * @returns {void}
+   */
+  track(action, category) {
```
Comment (latata) [inline app/services/smart-banner.js:96]: https://github.com/Wikia/mobile-wiki/blob/dev/app/components/site-head.js#L60

File: app/services/smart-banner.js [hunk 6]
```diff
@@ -36,21 +78,21 @@ export default Service.extend({
     };
 
     date.setTime(date.getTime() + (days * this.dayInMiliseconds));
-    window.Cookies.set(this.cookieName, 1, cookieOptions);
+    window.Cookies.set(cookieName, 1, cookieOptions);
   },
 
   isCookieSet() {
-    return window.Cookies.get(this.cookieName) === '1';
+    return window.Cookies.get(this.customCookieName) === '1' || window.Cookies.get(this.customCookieName) === '1';
```
Comment (latata) [inline app/services/smart-banner.js]: You check the same cookie twice ;)

File: app/templates/components/custom-smart-banner.hbs [hunk 7]
```diff
@@ -0,0 +1,29 @@
+<div class="fandom-app-smart-banner__close" role="button" {{action "close"}}>
```
Comment (latata) [inline app/templates/components/custom-smart-banner.hbs]: I would rename classes or if you want to have shared styles I would go with more generic class name.

**Review by latata — CHANGES_REQUESTED** [review]

File: app/services/smart-banner.js (hunk 5 above)
Comment (rybmat) [reply app/services/smart-banner.js:96]: done

File: app/components/application-wrapper.js (hunk 1 above)
Comment (rybmat) [reply app/components/application-wrapper.js]: I would be careful with renaming classes, who knows what adEng is using them for ;p

**Review by latata — APPROVED** [review]

**Pull request merged**